js/views/modals/purchase/Purchase.js

```diff
@@ -192,39 +200,38 @@ export default class extends BaseModal {
     this.close();
   }
 
-  clickModerated(e) {
-    const checked = $(e.target).prop('checked');
-    this.getCachedEl('.js-moderator').toggleClass('hide', !checked);
-    this.getCachedEl('.js-moderatorNote').toggleClass('hide', !checked);
-    this.order.moderated = checked;
-
-    if (checked) {
-      // re-select the previously selected moderator, if any
-      for (const mod of this.moderators.modCards) {
-        if (mod.model.id === this._oldMod || this.moderators.selectedIDs[0]) {
-          mod.changeSelectState('selected');
-          break;
-        }
-      }
-    } else {
-      // deselect all the moderators after storing any selected moderator
-      this._oldMod = this.moderators.selectedIDs[0];
-      this.moderators.noneSelected = true;
-      this.moderators.deselectOthers();
-      this.order.set('moderator', '');
-    }
-  }
+  clickDirectPurchase() {
+    if (!this.order.moderated) return;
 
-  moderationOn(bool) {
-    this.order.moderated = bool;
-    this.moderators.noneSelected = !bool;
-    this.setState({ showModerators: bool });
+    this.order.moderated = false;
+    this.moderators.deselectOthers();
+    this.render();
   }
 
-
   onNoValidModerators() {
     this.order.moderated = false;
-    this.setState({ noValidModerators: true });
+    this.render();
+  }
+
+  togVerifiedModerators(bool) {
+    // if hiding unverified mods, deselect the selected mod if it is unverified and switch to
+    // direct payment
```

Comment:
Hmmm... instead of kicking them back to a direct payment, should we maybe select nothing?

My concern is the user selects a mod and they may have no idea that toggling a filter kicks them back to a direct payment and they may unknowingly purchase something directly when they thought they had a mod (but maybe where just curious and wanted to see the list of verified mods).

Would love to hear @morebrownies thoughts on this when he comes back.

Reply:
The interaction here is pretty confusing, I agree. If you select an unverified mod and then hide the unverified mods, it looks like no mod is selected, so we don't want to keep the hidden mod selected if you click purchase.

But like you said, the user may not realize the unverified mod they selected was de-selected. 

Since the UI uses radio buttons (even if in the HTML they aren't really radio buttons) I think we should stick with how radio buttons work, which means one should always be selected. Or we could change to checkboxes or something else.

I could also see disabling the ability to hide unverified moderators if you have an unverified moderator selected as an option. Maybe that's the simplest?